src/components/HRInterface.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+const HRInterface = () => {
+  const [criteria, setCriteria] = useState([]);
+  const [newCriterion, setNewCriterion] = useState('');
+  const [weights, setWeights] = useState({});
+
+  const addCriterion = () => {
+    if (newCriterion.trim() !== '') {
+      setCriteria([...criteria, newCriterion]);
+      setNewCriterion('');
+    }
+  };
+
+  const removeCriterion = (index) => {
+    const updatedCriteria = criteria.filter((_, i) => i !== index);
+    setCriteria(updatedCriteria);
+  };
+
+  const updateWeight = (index, weight) => {
+    setWeights({ ...weights, [index]: weight });
+  };
```

Comment:
You could abstract out the logic of the HR interface using a custom React hook so that it's decoupled from the component. That makes it easier to test the logic without have to do component-level testing or end-to-end tests. In addition, if you do changes the UI it won't necessarily mess up your business logic either. So for instance with the criteria logic you could make the following React hook:

```typescript
import { useState } from 'react';
export const useCriterion = (intialCriteria=[], initialNewCriterion='') => {
  const [criteria, setCriteria] = useState(intialCriteria);
  const [newCriterion, setNewCriterion] = useState(initialNewCriterion);
  
  const addCriterion = () => {
    if (newCriterion.trim() !== '') {
      setCriteria([...criteria, newCriterion]);
      setNewCriterion('');
    }
  };

  const removeCriterion = (index) => {
    const updatedCriteria = criteria.filter((_, i) => i !== index);
    setCriteria(updatedCriteria);
  };
  return {criteria, newCriterion, setNewCriterion, addCriteria, removeCriteria};
}
```
And then the hook can be used in the component like:
```jsx
import { useCriterion } from "hooks/useCriterion";

const HRInterface = () => {
  const { criteria, newCriterion, setNewCriterion, addCriteria, removeCriteria } = useCriterion();
  ...
 }
```